### PR TITLE
Issue 63 segment metadata loader extra field

### PR DESCRIPTION
**Pull request by michael-mclawhorn**
Addressing https://github.com/yahoo/fili/issues/63

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,11 @@ Current
 
 - [Documentation for topN](https://github.com/yahoo/fili/pull/43)
 
+- [Broken segment metadata with Druid v0.9.1]](https://github.com/yahoo/fili/issues/63)
+  * Made `NumberedShardSpec` ignore unexpected properties during deserialization
+  * Added tests to DataSourceMetadataLoaderSpec to test the v.0.9.1 optional field 'shardSpec.partitionDimensions' on segment info JSON.
```
Comment (cdeszaq) [inline CHANGELOG.md]: Should move to "fixes", since it isn't really an addition

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,11 @@ Current
 
 - [Documentation for topN](https://github.com/yahoo/fili/pull/43)
 
+- [Broken segment metadata with Druid v0.9.1]](https://github.com/yahoo/fili/issues/63)
```
Comment (will-lauer) [inline CHANGELOG.md]: Double `]`

**Comment by will-lauer** [comment]
:+1:

**Comment by jacobtolar** [comment]
I assume this is the change you were talking about @michael-mclawhorn @cdeszaq . This doesn't address the issue I was having, so I'll keep my change.

**Pull request merged**